File: solutions/python/hamming/2/hamming.py
def distance(strand_a: str, strand_b:str) -> int:
    """Compares two strands of DNA and calculates the Hamming Distance
    
    :param strand_a: The first strand of DNA.
    :param strand_b: The second strand of DNA.
    :return: int - The Hamming Distance."""
    
    if len(strand_a) != len(strand_b): raise ValueError("Strands must be of equal length.")

    return sum(letter_a != letter_b for letter_a, letter_b in zip(strand_a, strand_b))

File: solutions/php/hamming/2/Hamming.php
<?php
declare(strict_types=1);

function distance(string $strandA, string $strandB): int
{
    if (strlen($strandA) != strlen($strandB)) throw new \InvalidArgumentException("DNA strands must be of equal length.");

    $count = 0;
    foreach (str_split($strandA) as $key => $letter) {
        if ($letter != str_split($strandB)[$key]) $count++;
    }
    
    return $count;
}
